packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -200,6 +202,12 @@ const SceneView = ({
       onAppear={onAppear}
       onDisappear={onDisappear}
       onDismissed={onDismissed}
+      onHeaderBackButtonClicked={() => {
+        navigation.dispatch({
+          ...StackActions.pop(),
+          source: route.key,
```

Comment:
This should also specify the `target` option. Let's move it to `NativeStackViewInner` component like the other callbacks

Reply:
Fixed in [8421534](https://github.com/react-navigation/react-navigation/pull/10354/commits/84215349631d8ef71fc9a5159b385ff32b170a47)